File: src/main/java/org/polyfrost/polyblur/config/PolyBlurConfig.java
package org.polyfrost.polyblur.config;

import org.polyfrost.oneconfig.api.config.v1.Config;
import org.polyfrost.oneconfig.api.config.v1.annotations.Dropdown;
import org.polyfrost.oneconfig.api.config.v1.annotations.Slider;
import org.polyfrost.oneconfig.api.config.v1.annotations.Switch;
import org.polyfrost.polyblur.blurs.phosphor.PhosphorBlur;

public class PolyBlurConfig extends Config {

    // Temporary!
    public boolean enabled = true;

    // @Info(
    //        text = "Phosphor / Moulberry blur will ONLY work if either Fast Render is disabled or Force Disable Fast Render is enabled.",
    //        size = 2,
    //        type = InfoType.WARNING
    //)
    private Runnable info = () -> { };

    @Switch(
            title = "Force Disable Fast Render"
    )
    public boolean forceDisableFastRender = true;

    @Dropdown(
            title = "Blur Mode",
            options = {
                    "Monkey Blur",
                    "Phosphor Blur",
                    "Moulberry Blur"
            }
    )
    public int blurMode = 1;

    @Slider(
        title = "Blur Strength",
        min = 1, max = 10,
        step = 1
    )
    public int strength = 3;

    public PolyBlurConfig() {
        super("polyblur.json", "/polyblur_dark.svg", "PolyBlur", Category.COMBAT);
        addCallback("strength", () -> {
            if (blurMode == 1 && enabled) {
                PhosphorBlur.reloadIntensity();
            }
        });
        addCallback("blurMode", PhosphorBlur::reloadBlur);
    }
}

File: src/main/kotlin/org/polyfrost/polyblur/PolyBlurConfig.kt
package org.polyfrost.polyblur

import org.polyfrost.oneconfig.api.config.v1.KtConfig
import org.polyfrost.polyblur.blurs.phosphor.PhosphorBlur

object PolyBlurConfig : KtConfig(
    id = "${PolyBlur.ID}.json",
    title = PolyBlur.NAME,
    category = Category.COMBAT,
    icon = "/polyblur_dark.svg"
) {

    var forceDisableFastRender by switch(def = true, name = "Force Disable Fast Render", description = "Forces OptiFine's Fast Render option to be disabled.")

    var enabled by switch(def = true, name = "Enabled")

    var mode by dropdown(options = arrayOf("Monkey Blur", "Phospor Blur", "Moulberry Blur"), def = 1, name = "Blur Mode")

    var strength by slider(min = 1f, max = 10f, def = 3f, name = "Blur Strength")

    init {
        addCallback("strength") {
            if (mode == 1) {
                PhosphorBlur.reloadBlur()
            }
        }

        addCallback("mode", PhosphorBlur::reloadIntensity)
    }

}
